pulp_2to3_migration/app/plugin/rpm/migrator.py

```diff
@@ -271,11 +271,15 @@ def relate_packages_to_group(self, group_dc):
         packages = group_dc.content.packages
         package_list = [pkg['name'] for pkg in packages]
         pulp2_repo_id = group_dc.extra_data.get('pulp2_repo_id')
-        pulp2_repo = Pulp2Repository.objects.get(pulp2_repo_id=pulp2_repo_id)
-        # all pulp2 unit_ids for rpm within the pulp2repo
-        unit_ids = Pulp2RepoContent.objects.filter(
-            pulp2_repository=pulp2_repo,
-            pulp2_content_type_id='rpm').values_list('pulp2_unit_id', flat=True).iterator()
+        pulp2_repo = Pulp2Repository.objects.filter(pulp2_repo_id=pulp2_repo_id).first()
+        # if pulp2_repo is found, find all pulp2 unit_ids for rpm/srpm within the pulp2_repo
+        # otherwise find all pulp2 unit_ids for rpm/srpm content
+        filters = {}
+        if pulp2_repo:
+            filters['pulp2_repository'] = pulp2_repo
+        filters['pulp2_content_type_id__in'] = ['rpm', 'srpm']
```

Comment:
pardon my ignorance - i did not know package groups can contain srpms, is this a valid comps.xml?https://repos.fedorapeople.org/pulp/pulp/fixtures/srpm-unsigned/repodata/

The side effect of this trade off is - if a repo would have foo.srpm and foo.rpm, both of them would appear in the relations for PackageGroup.

Reply:
No idea, but seems like it.

I think it's ok that both are there. There is no way to identify what was meant to be a part of the package group but having binary and source shouldn't hurt, imo.

We can ask on the #rpm channel or what was the channel where we had the opensuse nosrc/src discussion? @dralley

Reply:
The IRC channel was #rpm.org.  The #rpm-ecosystem channel has a few extra people in it but I'm not sure which is more applicable.